chann_test.go

```diff
@@ -258,7 +258,7 @@ func TestNonblockSelectRace(t *testing.T) {
 		}()
 		c2.In() <- 1
 		select {
-		case <-c1.Out():
+		case <-c2.Out():
```

Comment:
I can still reproduce the failure even if this is changed to `c2`. I think intrinsically this case cannot be fixed easily because we clearly violated the single-channel semantics.

Reply:
Indeed, it looks like it's hard to be completely fixed.